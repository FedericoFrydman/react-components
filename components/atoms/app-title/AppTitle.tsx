import { forwardRef, Ref } from 'react';
import cn from 'classnames';

import { Text, TextProps } from '../text/Text';
import './app-title.scss';

export interface AppTitleProps extends TextProps {
  /**
   * Text of the label
   */
  text?: string;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * The ref to the HTML DOM element
   */
  ref?: Ref<HTMLParagraphElement>;
}

const AppTitleComponent = ({ text, classNames = [], ...props }: AppTitleProps, ref: AppTitleProps['ref']) => (
  <Text {...props} ref={ref} text={text} classNames={['lsux-app-title', cn(classNames)]} />
);

export const AppTitle = forwardRef<HTMLParagraphElement, AppTitleProps>(AppTitleComponent);
