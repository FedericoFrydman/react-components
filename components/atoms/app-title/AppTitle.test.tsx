import { createRef } from 'react';
import { render, screen } from '@testing-library/react';
import { shallow, ShallowWrapper } from 'enzyme';
import cn from 'classnames';

import { AppTitle, AppTitleProps } from './AppTitle';

describe('AppTitle Component', () => {
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<AppTitle text="some text" classNames={['custom--class']} />);
    testId = 'test-text';
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('classNames', () => {
      describe('when provided classNames', () => {
        const classNames = ['my-class-1', 'my-class-2'];

        it('gives the text the provided classNames', () => {
          const text = renderLabel({ classNames });

          expect(text).toHaveClass(cn(classNames), { exact: false });
        });
      });
    });

    describe('text', () => {
      const text = 'testing text';

      it('is displayed when present', () => {
        renderLabel({ text });

        expect(screen.getByText(text)).toBeVisible();
      });
    });
  });

  describe('ref', () => {
    const ref = createRef<HTMLParagraphElement>();

    it('provides a ref to the text', () => {
      expect(renderLabel({ ref })).toEqual(ref.current);
    });
  });

  const renderLabel = (props: AppTitleProps = {}): HTMLElement => {
    render(<AppTitle {...props} data-testid={testId} />);

    return screen.getByTestId(testId);
  };
});
