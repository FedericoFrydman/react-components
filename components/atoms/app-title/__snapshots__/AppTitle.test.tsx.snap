// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle Component should render the component 1`] = `
<ForwardRef(TextComponent)
  classNames={
    Array [
      "lsux-app-title",
      "custom--class",
    ]
  }
  text="some text"
/>
`;
